uxarray/core/grid.py

```diff
@@ -283,6 +242,190 @@ def __from_ds__(self, dataset):
+        coords_str = ""
+        if "Mesh2_node_x" in self._ds:
+            coords_str += f"  * Mesh2_node_x: {self.Mesh2_node_x.shape}\n"
+            coords_str += f"  * Mesh2_node_y: {self.Mesh2_node_y.shape}\n"
+        if "Mesh2_face_x" in self._ds:
+            coords_str += f"  * Mesh2_face_x: {self.Mesh2_face_x.shape}\n"
+            coords_str += f"  * Mesh2_face_y: {self.Mesh2_face_y.shape} \n"
+
+        connectivity_heading = "Grid Connectivity Variables:\n"
+        connectivity_str = ""
+        if "Mesh2_face_nodes" in self._ds:
+            connectivity_str += f"  * Mesh2_face_nodes: {self.Mesh2_face_nodes.shape}\n"
+        if "Mesh2_edge_nodes" in self._ds:
+            connectivity_str += f"  * Mesh2_edge_nodes: {self.Mesh2_edge_nodes.shape}\n"
+        if "Mesh2_face_edges" in self._ds:
+            connectivity_str += f"  * Mesh2_face_edges: {self.Mesh2_face_edges.shape}\n"
+        connectivity_str += f"  * nNodes_per_face: {self.nNodes_per_face.shape}\n"
+
+        return prefix + original_grid_str + dims_heading + dims_str + coord_heading + coords_str + \
+            connectivity_heading + connectivity_str
+
+    @property
+    def parsed_attrs(self):
+        """Dictionary of parsed attributes from the source grid."""
+        return self._ds.attrs
+
+    @property
+    def Mesh2(self):
+        """UGRID Attribute ``Mesh2``, which indicates the topology data of a 2D
+        unstructured mesh."""
+        return self._ds[self.grid_var_names["Mesh2"]]
+
+    @property
+    def Mesh2_node_x(self):
+        """UGRID Coordinate Variable ``Mesh2_node_x``, which contains the
+        longitude of each node.
+
+        Dimensions (``nMesh2_node``)
+        """
+        return self._ds[self.grid_var_names["Mesh2_node_x"]]
+
+    @property
+    def Mesh2_face_x(self):
+        """UGRID Coordinate Variable ``Mesh2_face_x``, which contains the
+        longitude of each face center.
+
+        Dimensions (``nMesh2_face``)
+        """
+        if "Mesh2_face_x" in self._ds:
+            return self._ds["Mesh2_face_x"]
+        else:
+            return None
+
+    @property
+    def Mesh2_face_y(self):
+        """UGRID Coordinate Variable ``Mesh2_face_y``, which contains the
+        latitude of each face center.
+
+        Dimensions (``nMesh2_face``)
+        """
+        if "Mesh2_face_y" in self._ds:
+            return self._ds["Mesh2_face_y"]
+        else:
+            return None
+
+    @property
+    def Mesh2_node_y(self):
+        """UGRID Coordinate Variable ``Mesh2_node_y``, which contains the
+        latitude of each node.
+
+        Dimensions (``nMesh2_node``)
+        """
+        return self._ds[self.grid_var_names["Mesh2_node_y"]]
+
+    @property
+    def Mesh2_node_z(self):
+        """UGRID Coordinate Variable ``Mesh2_node_z``, which contains the
+        latitude of each node.
+
+        Dimensions (``nMesh2_node``)
+        """
+        if self.grid_var_names["Mesh2_node_z"] in self._ds:
```

Comment:
@philipc2 An important thing that needs to be checked is:

In our previous discussion during the meeting , We agree on the following:

1. The `Grid.Mesh2_node_x` and `Grid.Mesh2_node_y` will always store the longitude/latitude in degree

2. Inside our Grid object, We will always have the `Mesh2_node_cart_x`, `Mesh2_node_cart_y` and `Mesh2_node_cart_z` that store the Cartesian x, y, z coordinates.

3. And if the Grid object read-in longitude/latitude data, we will call the `Grid._populate_cartesian_xyz_coord` and 
If the Grid object read-in cartesian coordinates data, we will call the `Grid. _populate_lonlat_coord` to make sure each type of coordinates are represented correctly

The problem arises at the `Grid. _populate_lonlat_coord`: In our docstring, it states:

> *Helper function that populates the longitude and latitude and store
>       it into the Mesh2_node_x and Mesh2_node_y. This is called when the
>       input data has "Mesh2_node_x", "Mesh2_node_y", "Mesh2_node_z" in
>       meters. Since we want "Mesh2_node_x" and "Mesh2_node_y" always have the
>       "degree" units.*

**This means If the users pass in the Grid with valid values in `Mesh2_node_z`, we will interpret it as the Z-coordinates in the Cartesian coordinates** 

Therefore, I think we should have further thought about the compatibility between renaming/removing `mesh2_node_z` and the `Grid. _populate_lonlat_coord`. Since all our algorithm calculations are based on the cartesian coordinates and we don't want to convert these two coordinates back and forth, we need attribute storing the cartesian coord.

Reply:
This is a good catch. I'll keep `Mesh2_node_z` added as a placeholder for now, but I'll exclude it from the user API until we can resolve this.

Reply:
Thanks